### PR TITLE
5.6 ps bug1529885

### PR DESCRIPTION
**Pull request by ihanick**
merge to 5.6 for https://github.com/percona/percona-server/pull/323
https://bugs.launchpad.net/percona-server/+bug/1529885

http://jenkins.percona.com/job/percona-server-5.6-param/1077/

5.6 contains more sanity checks for new transaction log files and I have considered to delete logs for 4K block size with RESET CHANGED_PAGE_BITMAPS while changed pages tracking is disabled.

### CODE REVIEW

File: mysql-test/suite/innodb/t/percona_changed_page_bmp_flush.test [hunk 1]
```diff
@@ -66,18 +66,43 @@ INSERT INTO t1 VALUES (3, REPEAT("c", 20000));
...
 file_exists $BITMAP_FILE;
 --replace_regex /_[[:digit:]]+\.xdb$//
 list_files $MYSQLD_DATADIR ib_modified_log*;
+--remove_files_wildcard $MYSQLD_DATADIR ib_logfile*
 --enable_reconnect
---exec echo "restart:--innodb-track-changed-pages=1 --innodb-flush-method=ALL_O_DIRECT" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--exec echo "restart:--innodb-track-changed-pages=0 --innodb-log-block-size=4096 --innodb-flush-method=ALL_O_DIRECT" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--source include/wait_until_connected_again.inc
+
+RESET CHANGED_PAGE_BITMAPS;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/innodb/t/percona_changed_page_bmp_flush.test]: Maybe just call remove_files_wildcard for bitmaps next to where the log files are removed? This should save a server restart

File: mysql-test/suite/innodb/t/percona_changed_page_bmp_flush.test [hunk 2]
```diff
@@ -66,18 +66,43 @@ INSERT INTO t1 VALUES (3, REPEAT("c", 20000));
...
+
+--exec echo "wait" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--shutdown_server 10
+--source include/wait_until_disconnected.inc
+--exec echo "restart:--innodb-track-changed-pages=1 --innodb-log-block-size=4096 --innodb-flush-method=ALL_O_DIRECT" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
```
Comment (laurynas-biveinis) [inline mysql-test/suite/innodb/t/percona_changed_page_bmp_flush.test:81]: include/restart_mysqld.inc can be used here and elsewhere

File: mysql-test/suite/innodb/t/percona_changed_page_bmp_flush.test [hunk 3]
```diff
@@ -66,18 +66,43 @@ INSERT INTO t1 VALUES (3, REPEAT("c", 20000));
...
+--shutdown_server 10
+--source include/wait_until_disconnected.inc
+file_exists $BITMAP_FILE;
+--replace_regex /_[[:digit:]]+\.xdb$//
+list_files $MYSQLD_DATADIR ib_modified_log*;
+--enable_reconnect
+--exec echo "restart:--innodb-track-changed-pages=0 --innodb-log-block-size=4096 --innodb-flush-method=ALL_O_DIRECT" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--source include/wait_until_connected_again.inc
+
+RESET CHANGED_PAGE_BITMAPS;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/innodb/t/percona_changed_page_bmp_flush.test]: Same comment as above

**Comment by ihanick** [comment]
http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1084/
- fixed number of restarts, removing innodb logs and bitmaps twice: 
  before and after innodb-log-block-size=4096
- Can't use include/restart_mysqld.inc with new code:
  -- should remove innodb logs and changed pages bitmaps while server is down
  -- in 5.7 this test uses source include/shutdown_mysqld.inc and include/start_mysqld.inc, but in 5.6 there is no support for $restart_parameters variable in start_mysqld.inc

**Comment by laurynas-biveinis** [comment]
LGTM, please work on 5.7

**Comment by laurynas-biveinis** [comment]
Since the 5.7 fix is unrelated and being worked on under a separate bug, this needs a null merge to 5.7 now.

**Comment by ihanick** [comment]
Rebase 5.6 branch before last 5.7 split
http://jenkins.percona.com/job/percona-server-5.6-param/1152/

**Pull request merged**